rules/aip0233/resource_reference_type_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0233
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestResourceReferenceType(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName string
+		TypeName string
+		RefType  string
+		Response string
+		problems testutils.Problems
+	}{
+		{"ValidChildType", "library.googleapis.com/Book", "child_type", "BatchCreateBooksResponse", nil},
+		{"ValidChildTypeLRO", "library.googleapis.com/Book", "child_type", "google.longrunning.Operation", nil},
+		{"ValidType", "library.googleapis.com/Shelf", "type", "BatchCreateBooksResponse", nil},
+		{"InvalidType", "library.googleapis.com/Book", "type", "BatchCreateBooksResponse", testutils.Problems{{Message: "not a type"}}},
+		{"InvalidChildType", "library.googleapis.com/Shelf", "child_type", "BatchCreateBooksResponse", testutils.Problems{{Message: "child_type"}}},
+	}
+
+	// Run each test.
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			file := testutils.ParseProto3Tmpl(t, `
+				import "google/api/resource.proto";
+				import "google/longrunning/operations.proto";
+				service Library {
+					rpc BatchCreateBooks(BatchCreateBooksRequest) returns ({{ .Response }}) {
+						option (google.longrunning.operation_info) = {
+							response_type: "BatchCreateBooksResponse"
+							metadata_type: "BatchCreateBooksResponse"
+						};
+					}
```

Comment:
Same comment regarding `operation_info` in the body of tests that are not LROs as above.

Reply:
Done!